onnx/defs/tensor/defs.cc

```diff
@@ -286,69 +291,75 @@ ONNX_OPERATOR_SET_SCHEMA(
           // the dimensions we are setting outputShape in the outputProduct
           // variable. The outputProduct will potentially be used for inferring
           // a dimension marked -1.
-          auto* outputShape =
-              ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+          auto* outputShape = ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
           TensorShapeProto::Dimension* negativeOneDim = nullptr;
           const auto& dataInputTensorType = ctx.getInputType(0)->tensor_type();
-          std::vector<bool> unresolvedZeros(targetShape.size(), false);
+          std::vector<bool> unresolvedZeros(targetShapeProto.dim_size(), false);
           int64_t outputProduct = 1;
-          for (int i = 0; i < static_cast<int>(targetShape.size()); ++i) {
+          bool outputProductValid = true;
+          for (int i = 0; i < static_cast<int>(targetShapeProto.dim_size()); ++i) {
             // Add a new dimension to outputShape
             auto* new_dim = outputShape->add_dim();
-            if (targetShape[i] == -1) {
-              // Check if multiple -1's. If not, set negativeOneDim, marking
-              // this dimension to potentially be filled in later.
-              if (negativeOneDim) {
-                fail_shape_inference(
-                    "Target shape may not have multiple -1 dimensions.");
+            if (targetShapeProto.dim(i).has_dim_param()) {
+              new_dim->set_dim_param(targetShapeProto.dim(i).dim_param());
```

Comment:
There is a tricky problem here, in case the actual value at runtime happens to be either 0 or -1. In that case, propagating the dimension in the output-shape is problematic.